Apps/Sandcastle/gallery/Lighting.html

```diff
@@ -0,0 +1,172 @@
+var scratchMoonDirection = new Cesium.Cartesian3();
+
+function getMoonDirection(result) {
+    result = Cesium.defined(result) ? result : new Cesium.Cartesian3();
+    var icrfToFixed = scratchIcrfToFixed;
+    var date = viewer.clock.currentTime;
+    if (!Cesium.defined(Cesium.Transforms.computeIcrfToFixedMatrix(date, icrfToFixed))) {
+        Cesium.Transforms.computeTemeToPseudoFixedMatrix(date, icrfToFixed);
+    }
+    var moonPosition = Cesium.Simon1994PlanetaryPositions.computeMoonPositionInEarthInertialFrame(date, scratchMoonPosition);
+    Cesium.Matrix3.multiplyByVector(icrfToFixed, moonPosition, moonPosition);
+    var moonDirection = Cesium.Cartesian3.normalize(moonPosition, scratchMoonDirection);
+    return Cesium.Cartesian3.negate(moonDirection, result);
+}
+
+var directionalLight = new Cesium.DirectionalLight({
+    direction : new Cesium.Cartesian3(0.2454278300540191, 0.8842635425193919, 0.39729481195458805)
+});
+
+var flashlight = new Cesium.DirectionalLight({
+    direction : scene.camera.directionWC // Updated every frame
+});
+
+var moonLight = new Cesium.DirectionalLight({
+    direction : getMoonDirection(), // Updated every frame
+    color : new Cesium.Color(0.9, 0.925, 1.0),
+    intensity : 0.5
+});
+
+var sunLight = new Cesium.SunLight();
+
+var customColorLight = new Cesium.DirectionalLight({
+    direction : new Cesium.Cartesian3(-0.2454278300540191, 0.8842635425193919, 0.39729481195458805),
+    color : Cesium.Color.fromCssColorString('#deca7c')
+});
+
+scene.preRender.addEventListener(function(scene, time) {
+    if (scene.light === flashlight) {
+        scene.light.direction = Cesium.Cartesian3.clone(scene.camera.directionWC, scene.light.direction);
+    } else if (scene.light === moonLight) {
+        scene.light.direction = getMoonDirection(scene.light.direction);
+    }
+});
+
+viewer.entities.add({
+    position : Cesium.Cartesian3.fromRadians(-2.1463338399937277, 0.6677959688982861, 32.18991401746337),
+    model : {
+        uri : '../../SampleData/models/CesiumBalloon/CesiumBalloon.glb',
+        scale : 7.0
+    }
+});
+
+viewer.entities.add({
+    position : Cesium.Cartesian3.fromRadians(-2.14633449752228, 0.667796065242357, 24.47647034111423),
+    cylinder : {
+        length : 8.0,
+        topRadius : 2.0,
+        bottomRadius : 2.0,
+        material : Cesium.Color.WHITE
+    }
+});
+
+viewer.entities.add({
+    position : Cesium.Cartesian3.fromRadians(-2.1463332294173365, 0.6677959755384729, 26.2876064083145),
+    ellipsoid : {
+        radii : new Cesium.Cartesian3(2.5, 2.5, 2.5),
+        material : Cesium.Color.WHITE.withAlpha(0.5)
+    }
+});
+
+function setTime(iso8601) {
+    var currentTime = Cesium.JulianDate.fromIso8601(iso8601);
+    var endTime = Cesium.JulianDate.addDays(currentTime, 2, new Cesium.JulianDate());
+
+    viewer.clock.currentTime = currentTime;
+    viewer.timeline.zoomTo(currentTime, endTime);
+}
+
+function reset() {
```

Comment:
might be helpful to inline this function inside the different lighting selections below, to be as clear as possible which settings are turned on or off

Reply:
I think the current way might be clearer because `reset` sets it back to the defaults. It's a little easier to tell what has to be changed. I added a comment which might help.